packages/snaps-cli/src/cmds/eval/evalHandler.ts

```diff
@@ -11,6 +11,7 @@ export async function snapEval(argv: YargsArgs): Promise<boolean> {
     return true;
   } catch (err) {
     logError(`Snap evaluation error: ${err.message}`, err);
-    process.exit(1);
+    process.exitCode = 1;
```

Comment:
It would be better in this case to throw an error, and let this be caught by the top-level error handler.

We want to stop any further execution, but right now this won't do that in all cases. This function is called in `buildHandler.ts` where it proceeds to call `manifestHandler` after calling this, regardless of what this returns.

Reply:
The same problem affects many of the other changes in this PR as well, e.g. `initHandler`.

If we want to avoid a stack trace in these cases (i.e. maybe they are expected errors caused by the user or the snap, not bugs in the CLI itself), we can make a custom error class for runtime errors. Then we can look for it in the top-level `.fail` handler for the CLI, and only print the message there rather than the stack. Something like that.

Reply:
It may be possible to handle the stack trace removal directly in the `.fail` handler, the docs seem to indicate this: https://yargs.js.org/docs/#api-reference-failfn-boolean